app/frontend/src/features/messages/groupchats/AdminsDialog.tsx

```diff
@@ -0,0 +1,249 @@
+  memberIsAdmin: boolean;
+  setError: (value: string) => void;
+}) {
+  const classes = useMembersDialogStyles();
+
+  const isCurrentUser = useAuthContext().authState.userId === member.userId;
+
+  const queryClient = useQueryClient();
+  const clearError = () => setError("");
+  const handleError = (error: GrpcError) => setError(error.message);
+  const invalidate = () => {
+    queryClient.invalidateQueries(["groupChatMessages", groupChatId]);
+    queryClient.invalidateQueries(["groupChats"]);
+    queryClient.invalidateQueries(["groupChat", groupChatId]);
+  };
+
+  const makeAdmin = useMutation<Empty, GrpcError, void>(
+    () => service.conversations.makeGroupChatAdmin(groupChatId, member),
+    {
+      onMutate: clearError,
+      onSuccess: () => {
+        const previousGroupChat = queryClient.getQueryData<GroupChat.AsObject>([
+          "groupChat",
+          groupChatId,
+        ]);
+        const newAdminUserIdsList = previousGroupChat?.adminUserIdsList ?? [];
+        newAdminUserIdsList.push(member.userId);
+        queryClient.setQueryData(["groupChat", groupChatId], {
+          ...previousGroupChat,
+          adminUserIdsList: newAdminUserIdsList,
+        });
+        invalidate();
+      },
+      onError: handleError,
+    }
+  );
+  const removeAdmin = useMutation<Empty, GrpcError, void>(
+    () => service.conversations.removeGroupChatAdmin(groupChatId, member),
+    {
+      onMutate: clearError,
+      onSuccess: () => {
+        const previousGroupChat = queryClient.getQueryData<GroupChat.AsObject>([
+          "groupChat",
+          groupChatId,
+        ]);
+        const newAdminUserIdsList = previousGroupChat?.adminUserIdsList ?? [];
+        newAdminUserIdsList.splice(
+          newAdminUserIdsList.indexOf(member.userId),
+          1
+        );
+        queryClient.setQueryData(["groupChat", groupChatId], {
+          ...previousGroupChat,
+          adminUserIdsList: newAdminUserIdsList,
+        });
+        invalidate();
+      },
+      onError: handleError,
+    }
+  );
+
+  const handleMakeAdmin = () => makeAdmin.mutate();
+  const handleRemoveAdmin = () => removeAdmin.mutate();
+
+  return (
+    <ListItem dense className={classes.memberListItemContainer}>
+      {
+        //TODO: Colours
+        memberIsAdmin ? (
+          isCurrentUser ? (
+            <ConfirmationDialogWrapper
+              title="Step down as admin?"
+              message={
+                `Are you sure you want to stop being an admin ` +
+                `of this group chat? You will not be able to become ` +
+                `an admin again unless another admin adds you.`
+              }
+            >
+              <IconButton
+                aria-label="Remove as admin"
+                size="small"
+                loading={removeAdmin.isLoading}
+                onClick={handleRemoveAdmin}
+              >
+                <CloseIcon />
+              </IconButton>
+            </ConfirmationDialogWrapper>
```

Comment:
```suggestion
            <ConfirmationDialogWrapper
              title="Step down as admin?"
              message={
                `Are you sure you want to stop being an admin ` +
                `of this group chat? You will not be able to become ` +
                `an admin again unless another admin adds you.`
              }
              onConfirm={handleRemoveAdmin}
            >
              {(setIsOpen) => (
                <IconButton
                aria-label="Remove as admin"
                size="small"
                loading={removeAdmin.isLoading}
                onClick={() => setIsOpen(true)}
              >
                <CloseIcon />
              </IconButton>
              )}
            </ConfirmationDialogWrapper>
```
If using the render prop pattern I suggested in `ConfirmationDialogWrapper`